metrics_proto.go

```diff
@@ -562,3 +563,14 @@ func protoMetricDescriptorTypeToMetricKind(m *metricspb.Metric) (googlemetricpb.
 		return googlemetricpb.MetricDescriptor_METRIC_KIND_UNSPECIFIED, googlemetricpb.MetricDescriptor_VALUE_TYPE_UNSPECIFIED
 	}
 }
+
+func marshalMonitoredResource(mr *monitoredrespb.MonitoredResource) *monitoredrespb.MonitoredResource {
+	mb, me := proto.Marshal(mr)
```

Comment:
does Marshal cache the serialization internally?

Reply:
No :(. I did a big hack which proves to not help that much, and set the encoded bytes directly